java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/WikiService.java

```diff
@@ -86,10 +87,104 @@ public FeatureCollection getImagesById(long id, double lat, double lon) {
 		return new FeatureCollection(features.toArray(new Feature[0]));
 	}
 	
+	public String getWikiRawDataFromCache(String imageTitle, Long pageId) {
+		String query = "SELECT data FROM wiki.imagesrawdata WHERE title = ? AND pageId = ? LIMIT 1";
+		return jdbcTemplate.query(query, ps -> {
+			ps.setString(1, imageTitle);
+			ps.setLong(2, pageId);
+		}, rs -> {
+			if (rs.next()) {
+				return rs.getString("data");
+			}
+			return null;
+		});
+	}
+	
+	
+	public void saveWikiRawDataToCache(String imageTitle, Long pageId, String rawData) {
+		String query = "INSERT INTO wiki.imagesrawdata (pageId, title, data) VALUES (?, ?, ?)";
+		jdbcTemplate.update(query, ps -> {
+			ps.setLong(1, pageId);
+			ps.setString(2, imageTitle);
+			ps.setString(3, rawData);
+		});
+	}
+	
+	
+	public String parseRawImageInfo(String imageTitle) {
+		// https://dumps.wikimedia.org/commonswiki/
+		HttpURLConnection connection = null;
+		String encodedImageTitle = URLEncoder.encode(imageTitle, StandardCharsets.UTF_8);
+		String urlStr = "https://commons.wikimedia.org/wiki/File:" + encodedImageTitle + "?action=raw";
```

Comment:
Please implement strong validation of `imageTitle` to avoid arbitrary requests caused by user-defined parameter.